src/coreclr/src/jit/compiler.h

```diff
@@ -1614,9 +1624,24 @@ struct fgArgTabEntry
     }
 
     // Get the number of bytes that this argument is occyping on the stack.
-    unsigned GetStackSize() const
+    unsigned GetStackByteSize() const
     {
-        return (TARGET_POINTER_SIZE * this->numSlots);
+        if (IsHfaRegArg())
```

Comment:
It seems like this is redundant with the check below. You could still retain the assert below, but if `IsHfaRegArg()` returns true, wouldn't `(!IsSplit() && numRegs > 0)` also be true?